cfe_internal/enterprise/CFE_knowledge.cf

```diff
@@ -137,6 +137,14 @@ bundle agent cfe_internal_setup_knowledge
       create => "true",
       perms => mog("0570", "root", $(def.cf_apache_group) );
 
+      "$(cfe_internal_hub_vars.docroot)/application/cache/."
+       handle => "cfe_internal_setup_knowledge_files_doc_root_application_cache",
+       create => "true",
+       depth_search => recurse_basedir("inf"),
+       perms => mog("0660", $(def.cf_apache_user), $(def.cf_apache_group)),
```

Comment:
So, this promise targets both files and directories from the promised directory all the way down.

With the changes to rxdirs, you won't get execute (list) permission on directories. Is that OK or do you need to be able to list the files in the directories?

If so, then you should split this into 2 promises. One that targets files (file_select) and one that targets directories, each specifying the permissions you want for files or directories respectively.

Reply:
actually I don't need nested directories, only files, changed it.

Reply:
now it ```info: Owner of '/var/cfengine/httpd/htdocs/application/cache' was 0, set to 997
    info: Directory '/var/cfengine/httpd/htdocs/application/cache' had permissions 0550, changed it to 0770
    info: Owner of '/var/cfengine/httpd/htdocs/application/cache' was 997, set to 0
    info: Directory '/var/cfengine/httpd/htdocs/application/cache' had permissions 0770, changed it to 0550```
    
    changes perms back

Reply:
Ok, I pushed a couple changes to your fork, give that a go. It stopped the flipping for me.